server/index.js

```diff
@@ -24,14 +24,14 @@ const internalPrefixes = [
 ]
 
 export default class Server {
-  constructor ({ dir = '.', dev = false, staticMarkup = false, quiet = false } = {}) {
+  constructor ({ dir = '.', dev = false, staticMarkup = false, quiet = false, conf = {} } = {}) {
     this.dir = resolve(dir)
     this.dev = dev
     this.quiet = quiet
     this.router = new Router()
-    this.hotReloader = dev ? new HotReloader(this.dir, { quiet }) : null
+    this.hotReloader = dev ? new HotReloader(this.dir, { quiet, conf }) : null
     this.http = null
-    this.config = getConfig(this.dir)
+    this.config = getConfig(this.dir, conf)
```

Comment:
Well, for my use case, I still want to be able to read from filesystem to overwrite default & custom server.

That's why I have this order in Object.assign

Your thoughts?